src/test/java/com/spotify/ffwd/http/HttpClientTest.java

```diff
@@ -70,41 +72,36 @@ public void tearDown() {
 
     @Test
     public void testSendBatchSuccess() {
-        final String batchRequest =
-            "{\"commonTags\":{\"what\":\"error-rate\"},\"commonResource\":{},\"points\":"
-            + "[{\"key\":\"test_key\",\"tags\":{\"what\":\"error-rate\"},\"resource\":"
-            + "{},\"value\":1234.0,\"timestamp\":11111}]}";
-
-        final HttpRequest request = request()
-            .withMethod("POST")
-            .withPath("/v1/batch")
-            .withHeader("content-type", "application/json")
-            .withBody(batchRequest);
-
-        mockServerClient.when(request).respond(response().withStatusCode(200));
-
+        final String batchRequest = TestUtils.createJsonString(TestUtils.BATCH);
+        final HttpRequest request = sendRequest(batchRequest, RawHttpClient.V1_BATCH_ENDPOINT, SUCCESS);
         httpClient.sendBatch(TestUtils.BATCH).toCompletable().await();
-
         mockServerClient.verify(request, VerificationTimes.atLeast(1));
     }
 
     @Test
-    public void testSendBatchFail() {
-        expected.expectMessage("500: Internal Server Error");
-
-        final String batchRequest =
-            "{\"commonTags\":{\"what\":\"error-rate\"},\"commonResource\":{},\"points\":"
-            + "[{\"key\":\"test_key\",\"tags\":{\"what\":\"error-rate\"},\"resource\":"
-            + "{},\"value\":1234.0,\"timestamp\":11111}]}";
+    public void testSendBatchSuccessV2() {
+        final String batchRequest = TestUtils.createJsonString(TestUtils.BATCH_V2);
+        final HttpRequest request = sendRequest(batchRequest, RawHttpClient.V2_BATCH_ENDPOINT, SUCCESS);
+        httpClient.sendBatch(TestUtils.BATCH_V2).toCompletable().await();
+        mockServerClient.verify(request, VerificationTimes.atLeast(1));
+    }
 
+    private HttpRequest sendRequest(final String json, final String url, final int statusCode){
         final HttpRequest request = request()
-            .withMethod("POST")
-            .withPath("/v1/batch")
-            .withHeader("content-type", "application/json")
-            .withBody(batchRequest);
+                .withMethod("POST")
+                .withPath( "/" + url)
+                .withHeader("content-type", "application/json")
+                .withBody(json);
 
-        mockServerClient.when(request).respond(response().withStatusCode(500));
+        mockServerClient.when(request).respond(response().withStatusCode(statusCode));
+        return request;
+    }
 
+    @Test
+    public void testSendBatchFail() {
+        expected.expectMessage("500: Internal Server Error");
+        final String batchRequest = TestUtils.createJsonString(TestUtils.BATCH);
+        final HttpRequest request = sendRequest( batchRequest, RawHttpClient.V1_BATCH_ENDPOINT, 500);
```

Comment:
could also make a FAILURE = 500 like the success variable for consistency

Reply:
Sure

Reply:
Done